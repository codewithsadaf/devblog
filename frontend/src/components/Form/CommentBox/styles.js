//imports
import styled, { css } from "styled-components";
//components styles
import { InputComponent } from "../Input/styles";
import { ButtonContainer as Button } from "../../Layout/Button/styles";

export const Container = styled.div`
    display: flex;
    align-items: center;
    justify-content: space-between;
    width: 100%;

    & ${Button} {
        padding: 12px 16px;
    }
`;

export const CommentInput = styled(InputComponent)`
    ${({ theme }) => css`
        border: 1px solid ${theme.colors.gray};
        transition: ${theme.transition.tran_03};
        padding: 12px 7px;
        width: 90%;
        border-radius: 5px;
        font-size: 1em;
        margin-right: 7px;

        &:focus {
            border: 1px solid ${theme.colors.primaryColor};
            transition: ${theme.transition.tran_03};
        }
    `}
`;
