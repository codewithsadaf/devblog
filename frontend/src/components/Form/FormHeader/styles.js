//imports
import styled, { css } from "styled-components";
//components styles
import { TextComponent } from "../../Layout/TextComponent";

export const Container = styled.div`
    ${({ theme }) => css`
        margin-bottom: 1em;
    `}
`;

export const Text = styled(TextComponent)`
    margin-top: 0.3em;
`;
