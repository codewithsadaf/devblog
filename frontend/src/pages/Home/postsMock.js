export const postsMock = [
    {
        _id: "1",
        image: "example.png",
        title: "post title 1",
        content: "<p>post content</p>",
        user: {
            name: "john doe",
        },
        tags: ["#tag1", "#tag2"],
        createdAt: "2023-04-05T19:27:03.125+00:00",
        updatedAt: "2023-04-05T19:27:03.125+00:00",
    },
    {
        _id: "2",
        image: "example.png",
        title: "post title 2",
        content: "<p>post content</p>",
        user: {
            name: "john doe",
        },
        tags: ["#tag1", "#tag3"],
        createdAt: "2023-04-05T19:27:03.125+00:00",
        updatedAt: "2023-04-05T19:27:03.125+00:00",
    },
    {
        _id: "3",
        image: "example.png",
        title: "post title 3",
        content: "<p>post content</p>",
        user: {
            name: "john doe",
        },
        tags: ["#tag7", "#tag5"],
        createdAt: "2023-04-05T19:27:03.125+00:00",
        updatedAt: "2023-04-05T19:27:03.125+00:00",
    },
    {
        _id: "4",
        image: "example.png",
        title: "post title 4",
        content: "<p>post content</p>",
        user: {
            name: "Jane doe",
        },
        tags: ["#tag1", "#tag3"],
        createdAt: "2023-04-05T19:27:03.125+00:00",
        updatedAt: "2023-04-05T19:27:03.125+00:00",
    },
    {
        _id: "5",
        image: "example.png",
        title: "post title 5",
        content: "<p>post content</p>",
        user: {
            name: "Jane doe",
        },
        tags: ["#tag2", "#tag4"],
        createdAt: "2023-04-05T19:27:03.125+00:00",
        updatedAt: "2023-04-05T19:27:03.125+00:00",
    },
    {
        _id: "6",
        image: "example.png",
        title: "post title 6",
        content: "<p>post content</p>",
        user: {
            name: "Jane doe",
        },
        tags: ["#tag8", "#tag1"],
        createdAt: "2023-04-05T19:27:03.125+00:00",
        updatedAt: "2023-04-05T19:27:03.125+00:00",
    },
];
