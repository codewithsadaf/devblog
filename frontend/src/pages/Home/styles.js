//imports
import styled, { css } from "styled-components";

export const HomeContainer = styled.div`
    ${({ theme }) => css`
        width: 600px;
        margin: 3em auto 0;

        @media ${theme.media.lteMedium} {
            max-width: 100%;
        }
    `}
`;

export const MessageWrapper = styled.div`
    margin-top: 5em;
    text-align: center;
    display: flex;
    align-items: center;
    justify-content: center;
`;
