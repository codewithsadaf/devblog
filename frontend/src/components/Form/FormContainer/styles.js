//imports
import styled, { css } from "styled-components";
//components styles
import { ButtonContainer as Button } from "../../../components/Layout/Button/styles";

export const Container = styled.div`
    width: 100%;
    height: 100%;
    display: flex;
    align-items: center;
    justify-content: center;
`;

export const Form = styled.form`
    ${({ theme }) => css`
        width: 450px;
        margin-top: 3em;

        & > ${Button} {
            margin-top: 1em;
        }

        @media ${theme.media.lteMedium} {
            width: 100%;
            min-width: 0;
            max-width: 100%;
        }
    `}
`;
