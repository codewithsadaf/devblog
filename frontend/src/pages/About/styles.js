//imports
import styled, { css } from "styled-components";
//components styles
import { Text as TextComponent } from "../../components/Layout/TextComponent/styles";
import { Container as Heading } from "../../components/Layout/Heading/styles";

export const AboutContainer = styled.div`
    ${({ theme }) => css`
        max-width: 60%;
        margin: 3em auto 0;
        text-align: center;

        & > ${TextComponent} {
            margin-top: 0.5em;
        }

        @media ${theme.media.lteMedium} {
            max-width: 100%;

            & > ${Heading} {
                font-size: ${theme.font.sizes.xxlarge};
                line-height: ${theme.font.line_height.h3};
            }
        }
    `}
`;
