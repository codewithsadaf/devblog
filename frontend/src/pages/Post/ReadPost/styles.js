//imports
import styled, { css } from "styled-components";
import { Link } from "react-router-dom";
//components styles
import { ImageElm } from "../../../components/Layout/ImageBox/styles";
import { Text as TextComponent } from "../../../components/Layout/TextComponent/styles";

//PAGE CONTAINER
export const Container = styled.div`
    margin: 3em auto 0;
`;

//RETURN TO HOME TEXT
export const BackWrapper = styled.div`
    width: 100%;
    display: flex;
    align-items: center;
    justify-content: flex-start;
    margin-bottom: 1.2em;
    align-items: center;
`;

export const BackLink = styled(Link)`
    ${({ theme }) => css`
        color: ${theme.colors.primaryColor};
        transition: ${theme.transition.tran_03};
        display: flex;
        align-items: center;

        &:hover {
            transition: ${theme.transition.tran_03};
            opacity: 0.7;
        }
    `}
`;

//POST HEADER
export const PostHeader = styled.div`
    ${({ theme }) => css`
        width: 80%;
        margin: 0 auto;
        text-align: center;

        ${ImageElm} {
            height: 50%;
            width: 50%;
            margin: 2em auto;

            @media ${theme.media.lteMedium} {
                width: 100%;
                height: 100%;
            }
        }

        ${TextComponent} {
            margin: 1em 0;
        }

        @media ${theme.media.lteMedium} {
            width: 100%;
        }
    `}
`;

//POST CONTENT
export const ContentWrapper = styled.div`
    ${({ theme }) => css`
        width: 60%;
        margin: 0 auto;

        @media ${theme.media.lteMedium} {
            width: 100%;
        }
    `}
`;

export const DisplayContent = styled.div`
    ${({ theme }) => css`
        text-align: left;

        h1,
        h2,
        h3,
        h4,
        h5,
        h6 {
            margin-bottom: 0.5em;
            font-weight: bold;
        }

        p {
            font-size: 18px;
            margin-bottom: 1em;
        }

        a {
            font-weight: bold;
            color: ${theme.colors.primaryColor};
            text-decoration: underline;
        }

        ol,
        ul {
            margin-left: 2em;

            li {
                margin-bottom: 1em;
            }
        }

        ul {
            list-style-type: disc;
        }

        ol {
            list-style-type: decimal;
        }

        img {
            width: 100%;
            display: block;
            margin: 2em 0;
        }
    `}
`;

//POST TAGS
export const TagsWrapper = styled.div`
    display: flex;
    justify-content: center;
    align-items: center;
    width: 100%;
    margin: 3em 0;
    flex-wrap: wrap;
`;

export const TagText = styled.span`
    display: block;
    padding: 0 0.5em;
    font-weight: 500;

    &::before {
        content: "#";
        font-weight: bold;
    }
`;

export const TagsInfo = styled.span`
    display: block;
    margin-right: 0.5em;
    font-weight: bold;
`;

//POST LINE
export const Line = styled.div`
    ${({ theme }) => css`
        height: 2px;
        width: 100%;
        background-color: ${theme.colors.footerBackground};
        margin: 1em 0;
    `}
`;

//POST COMMENTS
export const CommentsNumberText = styled.h5`
    ${({ theme }) => css`
        font-size: ${theme.font.sizes.large};
        line-height: ${theme.font.line_height.h5};
    `}
`;

export const CommentsWrapper = styled.div`
    width: 100%;
    display: flex;
    flex-direction: column;
`;

export const SignUpWrapper = styled.div`
    margin: 2em 0;
    display: flex;
    align-items: center;
    justify-content: space-between;
    flex-direction: column;
    width: 100%;
    text-align: center;

    button {
        margin-top: 1em;
    }
`;

export const SignUpText = styled.h4`
    ${({ theme }) => css`
        font-size: ${theme.font.sizes.xlarge};
        line-height: ${theme.font.line_height.h4};
    `}
`;
