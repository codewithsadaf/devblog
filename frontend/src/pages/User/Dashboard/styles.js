//imports
import styled, { css } from "styled-components";
//components styles
import { ImageElm } from "../../../components/Layout/ImageBox/styles";
import { Container as Heading } from "../../../components/Layout/Heading/styles";
import { ButtonContainer as Button } from "../../../components/Layout/Button/styles";
import { InputComponent } from "../../../components/Form/Input/styles";
//react icons
import { FaCog } from "react-icons/fa";
import { Link } from "react-router-dom";

export const Container = styled.div`
    ${({ theme }) => css`
        width: 80%;
        margin: 3em auto 0;

        @media ${theme.media.lteMedium} {
            width: 100%;
        }
    `}
`;

export const TitleWrapper = styled.div`
    display: flex;
    align-items: center;
    justify-content: center;
    width: 100%;
    margin-bottom: 1.2em;
`;

export const ProfileWrapper = styled.div`
    ${({ theme }) => css`
        display: flex;
        align-items: center;
        justify-content: space-between;

        @media ${theme.media.lteMedium} {
            flex-direction: column;
        }
    `}
`;

export const ProfileImageWrapper = styled.div`
    ${({ theme }) => css`
        flex-basis: 40%;

        ${ImageElm} {
            width: 300px;
            height: 300px;
            object-fit: fill;
            border-radius: 50%;

            @media ${theme.media.lteMedium} {
                width: 150px;
                height: 150px;
            }
        }

        @media ${theme.media.lteMedium} {
            flex-basis: 100%;
        }
    `}
`;

export const ProfileInformationWrapper = styled.div`
    ${({ theme }) => css`
        flex-basis: 58%;
        min-height: 250px;
        text-align: left;

        ${Heading}:nth-of-type(2) {
            margin: 0.3em 0;
        }

        @media ${theme.media.lteMedium} {
            flex-basis: 100%;
            width: 100%;
            margin-left: 0;
        }
    `}
`;

export const RowStart = styled.div`
    display: flex;
    align-items: center;
    justify-content: flex-start;
    flex-wrap: wrap;
`;

export const CogLink = styled(Link)`
    ${({ theme }) => css`
        font-size: 1.2em;
        margin-left: auto;
        cursor: pointer;
        color: ${theme.colors.primaryColor};
        opacity: 1;
        transition: ${theme.transition.tran_05};

        &:hover {
            opacity: 0.6;
            transition: ${theme.transition.tran_05};
        }
    `}
`;

export const CogIcon = styled(FaCog)`
    ${({ theme }) => css``}
`;

export const PostsCommentsInfoWrapper = styled.div`
    ${({ theme }) => css`
        display: flex;
        width: 100%;
        border: 1px solid ${theme.colors.gray};
        border-radius: 7px;
        padding: 20px;
        margin-top: 1em;
    `}
`;

export const PostsInfoWrapper = styled.div`
    display: flex;
    flex-direction: column;
    align-items: center;
    margin-right: 2em;
`;

export const CommentsInfoWrapper = styled.div`
    display: flex;
    flex-direction: column;
    align-items: center;
`;

export const ViewOptionWrapper = styled.div`
    ${({ theme }) => css`
        margin: 2em 0;
        display: flex;
        border-bottom: 2px solid ${theme.colors.primaryColor};
    `}
`;

export const ViewOption = styled.span`
    ${({ theme }) => css`
        background-color: transparent;
        color: ${theme.colors.primaryColor};
        padding: 7px;
        font-size: ${theme.font.sizes.medium};
        border-radius: 7px 7px 0 0;
        font-weight: 500;
        cursor: pointer;
        transition: ${theme.transition.tran_03};

        &.active {
            background-color: ${theme.colors.primaryColor};
            color: ${theme.colors.white};
            transition: ${theme.transition.tran_03};
            pointer-events: none;
        }
    `}
`;

export const DashboardHeader = styled.div`
    ${({ theme }) => css`
        display: flex;
        justify-content: space-between;
        align-items: center;
        width: 100%;
        border-bottom: 2px solid ${theme.colors.gray};
        font-weight: 700;
        padding: 10px;
    `}
`;

export const DashboardText = styled.span`
    ${({ theme }) => css`
        font-size: ${theme.font.sizes.default};
        line-height: ${theme.font.line_height.default};
    `}
`;

export const DashboardRow = styled.div`
    ${({ theme }) => css`
        display: flex;
        justify-content: space-between;
        align-items: center;
        padding: 10px;
        width: 100%;
        border-bottom: 2px solid ${theme.colors.lightGray};

        & ${DashboardText} {
            display: block;
            width: 60%;
            word-wrap: break-word;

            @media ${theme.media.lteMedium} {
                width: 100%;
            }
        }

        @media ${theme.media.lteMedium} {
            flex-direction: column;
            justify-content: flex-start;
            align-items: flex-start;
        }
    `}
`;

export const DashboardActionsWrapper = styled.div`
    ${({ theme }) => css`
        margin-left: 2em;
        display: flex;
        justify-content: space-between;
        align-items: center;

        & ${Button} {
            margin-left: 0.5em;

            @media ${theme.media.lteMedium} {
                margin-left: 0;
                margin-right: 0.5em;
            }
        }

        @media ${theme.media.lteMedium} {
            margin-left: 0;
            margin: 1em 0;
        }
    `}
`;

export const CenterTextWrapper = styled.div`
    display: flex;
    align-items: center;
    justify-content: center;
    width: 100%;
    text-align: center;
    padding: 2em 0;
`;

export const EditInput = styled(InputComponent)`
    border: none;
    width: 100%;

    &:focus {
        border: none;
    }
`;
