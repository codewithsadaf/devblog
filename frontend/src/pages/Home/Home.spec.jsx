//imports
import { fireEvent, screen, waitFor } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";
import { act } from "react-dom/test-utils";
import userEvent from "@testing-library/user-event";
//styles
import { renderTheme } from "../../styles/render-theme";
//contexts
import { PostContext } from "../../context/PostContext";
//components
import { Home } from "./";
//mocks
import { postsMock } from "./postsMock";

//scrollTo Mock
Object.defineProperty(window.global, "scrollTo", { value: jest.fn() });

describe("<Home />", () => {
    let mockGetAllPosts;
    let mockIsLoading;

    beforeEach(() => {
        mockGetAllPosts = jest.fn();
        mockIsLoading = false;
    });

    describe("Rendering", () => {
        it("should render Home with 5 posts if no filter is applied", async () => {
            mockGetAllPosts.mockReturnValue(Promise.resolve(postsMock));
            renderTheme(
                <BrowserRouter>
                    <PostContext.Provider
                        value={{
                            isLoading: mockIsLoading,
                            getAllPosts: mockGetAllPosts,
                        }}
                    >
                        <Home />
                    </PostContext.Provider>
                </BrowserRouter>
            );

            //wait for GetAllPosts function to be called
            await waitFor(() => {
                expect(mockGetAllPosts).toHaveBeenCalledTimes(1);
            });

            //home heading
            expect(
                screen.getByRole("heading", {
                    name: "Veja os nossos posts mais recentes",
                })
            ).toBeInTheDocument();
            //search input
            expect(
                screen.getByPlaceholderText("Pesquisar por tags...")
            ).toBeInTheDocument();
            //search button
            expect(
                screen.getByRole("button", {
                    name: "Pesquisar",
                })
            ).toBeInTheDocument();
            //5 posts headings + home heading
            expect(screen.getAllByRole("heading", { level: 3 })).toHaveLength(
                6
            );
            //next page button
            expect(screen.getByLabelText("Next Page")).toBeInTheDocument();
            //pagination buttons
            expect(screen.getAllByLabelText("Pagination Button")).toHaveLength(
                2
            );
        });
        it("should render Home loading text if loading is true", async () => {
            mockGetAllPosts.mockReturnValue(Promise.resolve(postsMock));
            renderTheme(
                <BrowserRouter>
                    <PostContext.Provider
                        value={{
                            isLoading: true,
                            getAllPosts: mockGetAllPosts,
                        }}
                    >
                        <Home />
                    </PostContext.Provider>
                </BrowserRouter>
            );

            //wait for GetAllPosts function to be called
            await waitFor(() => {
                expect(mockGetAllPosts).toHaveBeenCalledTimes(1);
            });

            expect(
                screen.getByRole("heading", { name: "Carregando..." })
            ).toBeInTheDocument();
        });

        it("should render error text if fetch fails to get data", async () => {
            mockGetAllPosts.mockReturnValue(Promise.resolve(null));
            renderTheme(
                <BrowserRouter>
                    <PostContext.Provider
                        value={{
                            isLoading: false,
                            getAllPosts: mockGetAllPosts,
                        }}
                    >
                        <Home />
                    </PostContext.Provider>
                </BrowserRouter>
            );

            //wait for GetAllPosts function to be called
            await waitFor(() => {
                expect(mockGetAllPosts).toHaveBeenCalledTimes(1);
            });

            expect(
                screen.getByRole("heading", {
                    name: /Não foi possível carregar os posts no momento. Por favor, tente novamente mais tarde./,
                    level: 2,
                })
            ).toBeInTheDocument();
        });

        it("should render no posts found text if there are no posts", async () => {
            mockGetAllPosts.mockReturnValue(Promise.resolve([]));
            renderTheme(
                <BrowserRouter>
                    <PostContext.Provider
                        value={{
                            isLoading: false,
                            getAllPosts: mockGetAllPosts,
                        }}
                    >
                        <Home />
                    </PostContext.Provider>
                </BrowserRouter>
            );

            //wait for GetAllPosts function to be called
            await waitFor(() => {
                expect(mockGetAllPosts).toHaveBeenCalledTimes(1);
            });

            expect(
                screen.getByRole("heading", {
                    name: /Nenhum post encontrado/,
                    level: 2,
                })
            ).toBeInTheDocument();
        });
        it("should match snapshot", async () => {
            mockGetAllPosts.mockReturnValue(Promise.resolve(postsMock));
            const { container } = renderTheme(
                <BrowserRouter>
                    <PostContext.Provider
                        value={{
                            isLoading: false,
                            getAllPosts: mockGetAllPosts,
                        }}
                    >
                        <Home />
                    </PostContext.Provider>
                </BrowserRouter>
            );

            //wait for GetAllPosts function to be called
            await waitFor(() => {
                expect(mockGetAllPosts).toHaveBeenCalledTimes(1);
            });

            expect(container).toMatchSnapshot();
        });
    });
    describe("Functionality", () => {
        it("should update search input value on change", async () => {
            mockGetAllPosts.mockReturnValue(Promise.resolve(postsMock));
            renderTheme(
                <BrowserRouter>
                    <PostContext.Provider
                        value={{
                            isLoading: false,
                            getAllPosts: mockGetAllPosts,
                        }}
                    >
                        <Home />
                    </PostContext.Provider>
                </BrowserRouter>
            );

            //wait for GetAllPosts function to be called
            await waitFor(() => {
                expect(mockGetAllPosts).toHaveBeenCalledTimes(1);
            });

            const searchInput = screen.getByPlaceholderText(
                "Pesquisar por tags..."
            );

            fireEvent.change(searchInput, { target: { value: "test" } });

            expect(searchInput).toHaveValue("test");
        });
        it("should filter posts when user search for a single tag ", async () => {
            mockGetAllPosts.mockReturnValue(Promise.resolve(postsMock));
            renderTheme(
                <BrowserRouter>
                    <PostContext.Provider
                        value={{
                            isLoading: false,
                            getAllPosts: mockGetAllPosts,
                        }}
                    >
                        <Home />
                    </PostContext.Provider>
                </BrowserRouter>
            );

            //wait for GetAllPosts function to be called
            await waitFor(() => {
                expect(mockGetAllPosts).toHaveBeenCalledTimes(1);
            });

            const searchInput = screen.getByPlaceholderText(
                "Pesquisar por tags..."
            );

            const searchButton = screen.getByRole("button", {
                name: "Pesquisar",
            });

            fireEvent.change(searchInput, { target: { value: "tag1" } });

            act(() => {
                userEvent.click(searchButton);
            });

            await waitFor(() => {
                //it should display 4 posts with tag1 + home heading
                expect(
                    screen.getAllByRole("heading", { level: 3 })
                ).toHaveLength(5);
            });
        });
        it("should filter posts when user search for multiple tags ", async () => {
            mockGetAllPosts.mockReturnValue(Promise.resolve(postsMock));
            renderTheme(
                <BrowserRouter>
                    <PostContext.Provider
                        value={{
                            isLoading: false,
                            getAllPosts: mockGetAllPosts,
                        }}
                    >
                        <Home />
                    </PostContext.Provider>
                </BrowserRouter>
            );

            //wait for GetAllPosts function to be called
            await waitFor(() => {
                expect(mockGetAllPosts).toHaveBeenCalledTimes(1);
            });

            const searchInput = screen.getByPlaceholderText(
                "Pesquisar por tags..."
            );
            const searchButton = screen.getByRole("button", {
                name: "Pesquisar",
            });

            let searchedTags = "tag1 tag3";

            fireEvent.change(searchInput, { target: { value: searchedTags } });

            act(() => {
                userEvent.click(searchButton);
            });

            await waitFor(() => {
                //it should display (2) posts with tag1 and tag3 + home heading
                expect(
                    screen.getAllByRole("heading", { level: 3 })
                ).toHaveLength(3);
            });
        });

        it("should display error message on screen when no posts found with searched tag ", async () => {
            mockGetAllPosts.mockReturnValue(Promise.resolve(postsMock));
            renderTheme(
                <BrowserRouter>
                    <PostContext.Provider
                        value={{
                            isLoading: false,
                            getAllPosts: mockGetAllPosts,
                        }}
                    >
                        <Home />
                    </PostContext.Provider>
                </BrowserRouter>
            );

            //wait for GetAllPosts function to be called
            await waitFor(() => {
                expect(mockGetAllPosts).toHaveBeenCalledTimes(1);
            });

            const searchInput = screen.getByPlaceholderText(
                "Pesquisar por tags..."
            );
            const searchButton = screen.getByRole("button", {
                name: "Pesquisar",
            });

            let searchedTags = "potato";

            fireEvent.change(searchInput, { target: { value: searchedTags } });

            act(() => {
                userEvent.click(searchButton);
            });

            await waitFor(() => {
                expect(
                    screen.getByRole("heading", {
                        name: 'Nenhum post encontrado com a(s) tag(s) " #potato"',
                    })
                );
            });
        });
    });
});
