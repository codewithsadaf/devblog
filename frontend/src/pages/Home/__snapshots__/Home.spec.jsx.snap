// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> Rendering should match snapshot 1`] = `
.c0 {
  width: 600px;
  margin: 3em auto 0;
}

.c3 {
  text-transform: none;
  font-weight: bold;
  font-size: 2em;
  line-height: 110%;
}

.c8 {
  padding: 0.5em 1.8em;
  font-size: 1em;
  border: none;
  font-weight: bold;
  cursor: pointer;
  background: #000;
  color: #fff;
  border: 2px solid #000;
  border-radius: 4px;
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
}

.c8:hover {
  background: transparent;
  color: #000;
  border: 2px solid #000;
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
}

.c16 {
  padding: 0.5em 1.8em;
  font-size: 1em;
  border: none;
  font-weight: bold;
  cursor: pointer;
  background: transparent;
  color: #000;
  border: 2px solid #000;
  border-radius: 4px;
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
}

.c16:hover {
  background: #000;
  color: #fff;
  border: 2px solid #000;
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
}

.c5 {
  padding: 7px 0;
  outline: none;
  border: none;
  border-bottom: 1px solid #ccc;
  background-color: transparent;
  font-size: 1em;
  -webkit-transition: all 500ms ease-in-out;
  transition: all 500ms ease-in-out;
}

.c5:focus {
  border-bottom: 1px solid #000;
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
}

.c1 {
  width: 100%;
  margin: 0 auto;
  margin-bottom: 2em;
}

.c1 .c2 {
  margin-bottom: 0.5em;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  width: 90%;
  margin: 0 auto;
}

.c4 .c7 {
  width: 20%;
  padding: 7px;
  border-radius: 2px;
}

.c6 {
  width: 78%;
}

.c9 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  width: 100%;
}

.c17 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  margin-top: 1em;
}

.c18 {
  padding: 7px;
  height: 40px;
  width: 40px;
  border-radius: 4px;
  text-align: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  font-size: 1.2em;
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
  cursor: pointer;
  margin-right: 0.3em;
  background: transparent;
  color: #000;
  border: 1px solid #000;
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
}

.c18.active {
  background: #000;
  color: #fff;
  cursor: default;
  pointer-events: none;
}

.c18:hover {
  background: #000;
  color: #fff;
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
}

.c19 {
  background: #000;
  padding: 7px;
  height: 40px;
  width: 40px;
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
  background: transparent;
  color: #000;
  border: 1px solid #000;
  border-radius: 4px;
  text-align: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  font-size: 30px;
  cursor: pointer;
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
}

.c19:hover {
  background: #000;
  color: #fff;
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
}

.c19:first-child {
  margin-right: 0.3em;
}

.c13 {
  font-size: 1em;
  line-height: 150%;
}

.c10 {
  width: 100%;
  margin: 0 auto 3em;
  max-width: 600px;
}

.c10 .c12 {
  color: #444;
  font-size: 0.8em;
  line-height: 150%;
  font-style: italic;
}

.c11 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  margin-bottom: 1em;
  border-radius: 4px;
}

.c14 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin: 0.8em 0;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

.c15 {
  display: block;
  margin-right: 0.5em;
  font-weight: 500;
}

.c15::before {
  content: "#";
  font-weight: bold;
}

@media (max-width:768px) {
  .c0 {
    max-width: 100%;
  }
}

@media (max-width:768px) {
  .c4 .c7 {
    width: 35%;
  }
}

@media (max-width:768px) {
  .c4 {
    width: 100%;
  }
}

@media (max-width:768px) {
  .c6 {
    width: 60%;
  }
}

@media (max-width:768px) {
  .c10 .c2 {
    font-size: 1.7em;
    line-height: 120%;
  }
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <h3
        class="c2 c3"
      >
        Veja os nossos posts mais recentes
      </h3>
      <div
        class="c4"
      >
        <input
          class="c5 c6"
          placeholder="Pesquisar por tags..."
          type="text"
        />
        <button
          class="c7 c8"
          type="button"
        >
          Pesquisar
        </button>
      </div>
    </div>
    <div
      class="c9"
    >
      <div
        class="c10"
      >
        <img
          alt="post title 1"
          class="c11"
          src="example.png"
        />
        <h3
          class="c2 c3"
        >
          post title 1
        </h3>
        <p
          class="c12 c13"
        >
          Por: 
          john doe
           - 
          5 de Abril 2023
        </p>
        <div
          class="c14"
        >
          <span
            class="c15"
          >
            tag1
          </span>
          <span
            class="c15"
          >
            tag2
          </span>
        </div>
        <button
          class="c7 c16"
          type="button"
        >
          Ler
        </button>
      </div>
      <div
        class="c10"
      >
        <img
          alt="post title 2"
          class="c11"
          src="example.png"
        />
        <h3
          class="c2 c3"
        >
          post title 2
        </h3>
        <p
          class="c12 c13"
        >
          Por: 
          john doe
           - 
          5 de Abril 2023
        </p>
        <div
          class="c14"
        >
          <span
            class="c15"
          >
            tag1
          </span>
          <span
            class="c15"
          >
            tag3
          </span>
        </div>
        <button
          class="c7 c16"
          type="button"
        >
          Ler
        </button>
      </div>
      <div
        class="c10"
      >
        <img
          alt="post title 3"
          class="c11"
          src="example.png"
        />
        <h3
          class="c2 c3"
        >
          post title 3
        </h3>
        <p
          class="c12 c13"
        >
          Por: 
          john doe
           - 
          5 de Abril 2023
        </p>
        <div
          class="c14"
        >
          <span
            class="c15"
          >
            tag7
          </span>
          <span
            class="c15"
          >
            tag5
          </span>
        </div>
        <button
          class="c7 c16"
          type="button"
        >
          Ler
        </button>
      </div>
      <div
        class="c10"
      >
        <img
          alt="post title 4"
          class="c11"
          src="example.png"
        />
        <h3
          class="c2 c3"
        >
          post title 4
        </h3>
        <p
          class="c12 c13"
        >
          Por: 
          Jane doe
           - 
          5 de Abril 2023
        </p>
        <div
          class="c14"
        >
          <span
            class="c15"
          >
            tag1
          </span>
          <span
            class="c15"
          >
            tag3
          </span>
        </div>
        <button
          class="c7 c16"
          type="button"
        >
          Ler
        </button>
      </div>
      <div
        class="c10"
      >
        <img
          alt="post title 5"
          class="c11"
          src="example.png"
        />
        <h3
          class="c2 c3"
        >
          post title 5
        </h3>
        <p
          class="c12 c13"
        >
          Por: 
          Jane doe
           - 
          5 de Abril 2023
        </p>
        <div
          class="c14"
        >
          <span
            class="c15"
          >
            tag2
          </span>
          <span
            class="c15"
          >
            tag4
          </span>
        </div>
        <button
          class="c7 c16"
          type="button"
        >
          Ler
        </button>
      </div>
    </div>
    <div
      class="c17"
    >
      <div
        aria-label="Pagination Button"
        class="c18 active"
      >
        1
      </div>
      <div
        aria-label="Pagination Button"
        class="c18"
      >
        2
      </div>
      <div
        aria-label="Next Page"
        class="c19"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 0h24v24H0z"
            fill="none"
          />
          <path
            d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
          />
        </svg>
      </div>
    </div>
  </div>
</div>
`;
